File: Lab2/src/Model/PrintStmt.java
package Model;

import Exceptions.*;

public class PrintStmt implements IStmt {
    private Exp exp;

    public PrintStmt(VarExp v) {
        this.exp = v;
    }

    public String toString() {
        return "print(" + exp.toString() + ")";
    }

    public PrgState execute(PrgState state) throws DivisionByZero, VariableNotFound, OperatorNotFound {
        MyIList<Integer> out = state.getOut();
        MyIDictionary<String, Integer> symTbl = state.getSymTable();
        int val = exp.eval(symTbl);
        out.add(val);
        return state;
    }
}

File: A7/ToyLanguage/Model/PrintStmt.cs
using System;
namespace ToyLanguage.Model
{
    public class PrintStmt : IStmt
    {
        private Exp exp;

        public PrintStmt(Exp v)
        {
            this.exp = v;
        }
        public PrgState execute(PrgState state)
        {
            IList<int> output = state.getOutput();
            IDict<String, int> symTbl = state.getSymTbl();
            int val = exp.eval(symTbl);
            output.add(val);
            return null;
        }
        public override string ToString()
        {
            return "print(" + exp.ToString() + ")";
        }
    }
}
